analyzers/tests/SonarAnalyzer.UnitTest/TestCases/MemberInitializedToDefault.CSharp9.cs

```diff
@@ -1,15 +1,23 @@
-﻿record NativeInts
+﻿using System;
+
+record NativeInts
 {
     public const nint myConst = 0;  // Compliant
-    public nint field1 = 0;         // FN
+    public static nint staticField = 0; // Noncompliant
+    public nint field1 = 0;         // Noncompliant
     public nint field2 = 42;
-    public nuint field3 = 0;        // FN
+    public nuint field3 = 0;        // Noncompliant
     public nuint field4 = 42;
+    public nint field5 = IntPtr.Zero;         // Noncompliant
+    public nint field6 = 0x0000000000000000;  // Noncompliant
+    public nuint field7 = UIntPtr.Zero;       // Noncompliant
+    public nint field8 = new IntPtr(0);       // Noncompliant
+    public nint field9 = new IntPtr(staticField);
 
-    public nint Property1 { get; set; } = 0;    // FN
+    public nint Property1 { get; set; } = 0;    // Noncompliant
```

Comment:
[coverage]
to try and force [the property symbol to be null](https://sonarcloud.io/code?id=sonaranalyzer-dotnet&pullRequest=4717&selected=sonaranalyzer-dotnet%3Aanalyzers%2Fsrc%2FSonarAnalyzer.CSharp%2FRules%2FMemberInitializedToDefault.cs&line=67) you could add a test with an UnknownType property (which can happen in SonarLint when coding in the IDE) and see if the returned symbol is null

Reply:
This flow is a bit harder to reproduce since `GetDeclaredSymbol` will not return null but `NativeInts.PropertyName` in this case.